includes/functions.php

```diff
@@ -107,6 +107,22 @@ function getHostOS($device)
         }
     }
 
+    $pattern = $config['install_dir'] . '/includes/definitions/*.yaml';
+    foreach (glob($pattern) as $file) {
+        $tmp = Symfony\Component\Yaml\Yaml::parse(
+            file_get_contents($file)
+        );
+        if ($tmp['discovery']['method'] === 'sysObjectId' && is_array($tmp['discovery']['sysObjectId'])) {
+            if (starts_with($sysObjectId, $tmp['discovery']['sysObjectId'])) {
+                return $tmp['os'];
+            }
+        } elseif ($tmp['discovery']['method'] === 'sysDescr' && is_array($tmp['discovery']['sysDescr'])) {
```

Comment:
Some os require both sysObjectId and sysDescr checks, some allow either or it would be nice to accommodate that somehow.